lib/auth/auth.go

```diff
@@ -2439,6 +2439,28 @@ func (a *Server) CreateAccessRequest(ctx context.Context, req types.AccessReques
 	return nil
 }
 
+func (a *Server) DeleteAccessRequest(ctx context.Context, name string) error {
+	if err := a.DynamicAccessExt.DeleteAccessRequest(ctx, name); err != nil {
+		return trace.Wrap(err)
+	}
+	if err := a.emitter.EmitAuditEvent(ctx, &apievents.AccessRequestDelete{
+		Metadata: apievents.Metadata{
+			Type: events.AccessRequestDeleteEvent,
+			Code: events.AccessRequestDeleteCode,
+		},
+		UserMetadata: apievents.UserMetadata{
+			User:         ClientUsername(ctx),
+			Impersonator: ClientImpersonator(ctx),
+		},
+		ResourceMetadata: apievents.ResourceMetadata{
+			Name: name,
```

Comment:
All other field uses `RequestID:    params.RequestID` Can we align the `AccessRequestDelete` struct and use `RequestID` field instead of ResourceMetadata.Name for consistency ?

Reply:
I agree with you here; some resource types always use `ResourceMetadata` for all events, while other types always use a custom field - no resource type uses a mix, so to be consistent we should do what we already do for the `AccessRequestCreate` event.